common/persistence/versionhistory/version_history.go

```diff
@@ -192,9 +141,9 @@ func FindLCAItem(
 
 		if localVersionItem.Version == remoteVersionItem.Version {
 			if localVersionItem.GetEventId() > remoteVersionItem.GetEventId() {
-				return DuplicateItem(remoteVersionItem), nil
+				return remoteVersionItem, nil
 			}
-			return DuplicateItem(localVersionItem), nil
+			return localVersionItem, nil
```

Comment:
could you keep the original logic?

Reply:
Why? I can sort of understand when input parameters are getting copied, but what is the reason of coping return value? Original logic copied those items two times. Why? Just in case?

Reply:
as long as the returned value is not shared by multiple place, then ok.